pkg/config/config.go

```diff
@@ -56,6 +56,12 @@ func (c *Config) ApplyDefaults() error {
 		c.Integrations.ServerUsingTLS = c.Server.HTTPTLSConfig.TLSKeyPath != "" && c.Server.HTTPTLSConfig.TLSCertPath != ""
 	}
 
+	// The prometheus block remote write should trickle down to the integrations block if it is NOT defined.
+	promAndIntegrationsEnabled := c.Integrations.Enabled && c.Prometheus.Enabled
+	promHasRemoteIntegrationDoesNot := len(c.Prometheus.RemoteWrite) > 0 && len(c.Integrations.PrometheusRemoteWrite) == 0
+	if promAndIntegrationsEnabled && promHasRemoteIntegrationDoesNot {
+		c.Integrations.PrometheusRemoteWrite = c.Prometheus.RemoteWrite
+	}
```

Comment:
I think we can probably make some more assumptions here. If you move all of this inside `if c.IntegrationsEnabled` on line 53, you can just implement it as:

```go
if (c.Integrations.PrometheusRemoteWrite) == 0 {
  c.Integrations.PrometheusRemoteWrite = c.Prometheus.RemoteWrite 
}
```